include/mvd/bits/sonata_misc.hpp

```diff
@@ -71,24 +73,68 @@ inline Positions SonataFile::getPositions(const Range & range) const{
 }
 
 inline Rotations SonataFile::getRotations(const Range & range) const{
-    Positions res{boost::extents[range.count > 0 ? range.count : size_][4]};
+    using Quaternion = boost::math::quaternion<double>;
 
-    auto xs = pop_->getAttribute<double>("orientation_x", select(range, size_));
-    auto ys = pop_->getAttribute<double>("orientation_y", select(range, size_));
-    auto zs = pop_->getAttribute<double>("orientation_z", select(range, size_));
-    auto ws = pop_->getAttribute<double>("orientation_w", select(range, size_));
+    Positions res{boost::extents[range.count > 0 ? range.count : size_][4]};
 
-    // No direct slicing write access from std::vector
-    for (size_t i = 0; i < (range.count > 0 ? range.count : size_); ++i) {
-        res[i][0] = xs[i];
-        res[i][1] = ys[i];
-        res[i][2] = zs[i];
-        res[i][3] = ws[i];
+    const auto attrs = pop_->attributeNames();
```

Comment:
I would split this routine. At least quat = .... is done twice with the same code